rosbag2_compression/src/rosbag2_compression/sequential_compression_reader.cpp

```diff
@@ -142,6 +142,24 @@ std::vector<rosbag2_storage::TopicMetadata> SequentialCompressionReader::get_all
   throw std::runtime_error{"Bag is not open. Call open() before reading."};
 }
 
+void SequentialCompressionReader::set_filter(
+  const rosbag2_storage::StorageFilter & storage_filter)
+{
+  if (storage_) {
+    storage_->set_filter(storage_filter);
+    return;
+  }
+  throw std::runtime_error("Bag is not open. Call open() before setting "
+          "filter.");
```

Comment:
total bike shedding, but I somehow feel it reads easier if you invert the logic:

```
if (nullptr == storage_) {
  throw ...
}

storage_->set_filter(storage_filter);
```

same below

Reply:
So here I just copied the pattern in the three existing functions above these lines...